scipy/signal/sigtoolsmodule.c

```diff
@@ -1011,6 +1055,15 @@ PyObject *PyArray_OrderFilterND(PyObject *op1, PyObject *op2, int order) {
 
 fail:
 	if (zptr) PyDataMem_FREE(zptr);
+	if (b_ind)	 free(b_ind);
```

Comment:
These variables need to be initialized to NULL on top of the routine, otherwise this'll try to free uninitialized pointers.
Please check this also on the other changes.